backend/curation/api/v1/curation/datasets/actions.py

```diff
@@ -1,16 +1,28 @@
 from flask import jsonify, make_response
 
+from backend.common.utils.http_exceptions import InvalidParametersHTTPException
 from backend.curation.api.v1.curation.collections.common import extract_doi_from_links, reshape_dataset_for_curation_api
 from backend.portal.api.providers import get_business_logic
 
 
-def get():
+def get(schema_version: str = None):
     """
     Datasets index endpoint to retrieve full metadata for all public Datasets.
     """
-    collections_with_datasets = get_business_logic().get_all_mapped_collection_versions_with_datasets()
-
     all_datasets_with_collection_name_and_doi = []
+    if not schema_version:
+        collections_with_datasets = get_business_logic().get_all_mapped_collection_versions_with_datasets()
+    else:
+        version_parts = schema_version.split(".")
+        if len(version_parts) > 3 or not all(part.isdigit() for part in version_parts):
+            raise InvalidParametersHTTPException(detail="Invalid Schema Version Input")
+        while len(version_parts) < 3:
+            # wildcard match for exactly 1 character
```

Comment:
Add a test for this to make sure it escapes correctly, and the wild card filter works.